File: src/main/java/com/dude/dms/ui/components/standard/DmsDatePicker.java
package com.dude.dms.ui.components.standard;

import com.vaadin.flow.component.datepicker.DatePicker;

import java.util.Locale;

import static com.dude.dms.backend.brain.OptionKey.LOCALE;

public class DmsDatePicker extends DatePicker {

    public DmsDatePicker() {
        this("");
    }

    public DmsDatePicker(String label) {
        super(label);
        setLocale(Locale.forLanguageTag(LOCALE.getString()));
    }
}

File: src/main/java/com/dude/dms/ui/components/standard/DmsDatePicker.kt
package com.dude.dms.ui.components.standard

import com.dude.dms.brain.options.Options
import com.vaadin.flow.component.datepicker.DatePicker
import java.util.*

class DmsDatePicker(label: String = "") : DatePicker(label) {

    init {
        locale = Locale.forLanguageTag(Options.get().view.locale)
    }
}